graders/c/cgrader/cgrader.py

```diff
@@ -532,29 +572,31 @@ def add_manual_grading(self, points=1, name=None, description=None):
 
     def add_test_result(
         self,
-        name,
-        description="",
-        points=True,
-        msg="",
-        output="",
-        max_points=1,
-        field=None,
-        images=None,
-    ):
-        if not isinstance(points, int | float):
+        name: str,
+        description: str = "",
+        points: bool | float = True,
+        msg: str | None = "",
+        output: str = "",
+        max_points: float = 1,
+        field: str | None = None,
+        images: str | list[str] | dict[str, str] | list[dict[str, str]] | None = None,
```

Comment:
Nit (just to avoid issues with a potential case of a list containing both types of image representations, which should be valid though uncommon):
```suggestion
        images: str | dict[str, str] | list[str | dict[str, str]] | None = None,
```